itest/send_test.go

```diff
@@ -621,30 +622,90 @@ func testReattemptFailedReceiveUniCourier(t *harnessTest) {
 		wg   sync.WaitGroup
 	)
 
-	// Make a new node which will send the asset to the primary tapd node.
-	// We expect this node to fail because our send call will time out
-	// whilst the porter continues to attempt to send the asset.
-	sendTapd := setupTapdHarness(
+	// This tapd node will send the asset to the receiving tapd node.
+	// It will also transfer proof the related transfer proofs to the
+	// proof courier.
+	sendTapd := t.tapd
+
+	// Initialise a receiver tapd node. This node will attempt to retrieve
+	// the transfer proofs from the proof courier.
+	receiveTapd := setupTapdHarness(
 		t.t, t, t.lndHarness.Bob, t.universeServer,
 		func(params *tapdHarnessParams) {
 			params.expectErrExit = true
+			params.proofSendBackoffCfg = &proof.BackoffCfg{
+				BackoffResetWait: 1 * time.Second,
+				NumTries:         200,
+				InitialBackoff:   1 * time.Second,
+				MaxBackoff:       1 * time.Second,
+			}
 		},
 	)
 
-	// Use the primary node as the receiver node.
-	receiveTapd := t.tapd
+	// Mint an asset for sending using the sending tapd node.
+	rpcAssets := MintAssetsConfirmBatch(
+		t.t, t.lndHarness.Miner.Client, sendTapd,
+		[]*mintrpc.MintAssetRequest{simpleAssets[0]},
+	)
+
+	genInfo := rpcAssets[0].AssetGenesis
+
+	// Synchronize the Universe state of the second node, with the receiver
+	// node.
+	t.syncUniverseState(sendTapd, receiveTapd, len(rpcAssets))
+
+	// Create a new address for the receiver node.
+	recvAddr, err := receiveTapd.NewAddr(ctxb, &taprpc.NewAddrRequest{
+		AssetId: genInfo.AssetId,
+		Amt:     10,
+	})
+	require.NoError(t.t, err)
+	AssertAddrCreated(t.t, receiveTapd, rpcAssets[0], recvAddr)
+
+	// Stop receiving tapd node to simulate an offline receiver.
+	t.Logf("Stopping the receiving tapd node")
+	require.NoError(t.t, receiveTapd.stop(false))
+
+	// Send asset and then mine to confirm the associated on-chain tx.
+	sendAssetsToAddr(t, sendTapd, recvAddr)
+	_ = MineBlocks(t.t, t.lndHarness.Miner.Client, 1, 1)
+
+	// At this point, the proof courier service is running. We will
+	// therefore pause to allow the sender to transfer the proof to the
+	// proof courier service.
+	time.Sleep(2 * time.Second)
```

Comment:
Can we instead hook this into the sender side events to eliminate the sleep?

Reply:
Also use smaller timeouts for `BackoffResetWait` and `InitialBackoff` to drop total test time? Or is there some floor.

Reply:
Actually those don't seem to be the blockers for test time, I think it may be the receiver-side settings? And the params are applying to proof send not recv. Not sure.

Reply:
I think we could listen for a `SendStateComplete` state event from the sender node's chain porter. But I don't think it's ideal. Ideally we'd have a "proof delivered" event which contained the right data to tie it back to the particular send we're interested in.

I think there's room to add more event listening/assertion itest helper functions also. Listening for an event requires many lines of code currently.

Maybe we circle back once we've got better events in place? I can add a TODO if you agree.

Reply:
> Using timeouts of:

@jharveyb doesn't always pass on my computer with those numbers. I think they're cutting it too close.